src/components/Chat/ChatBodyComponent.tsx

```diff
@@ -1,20 +1,61 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
 import { ITextMessage } from '../../interfaces';
 import ChatMessageComponent from './ChatMessageComponent';
+import { Modal } from '../../components';
+import { SocketContext } from '../../providers';
+import { createGetAvailableQueueMessage, createJoinChatMessage } from '../../services';
+import { ModalContext } from '../../providers/ModalProvider';
 
 interface IProps {
   messages: ITextMessage[];
 }
 
 const ChatBodyComponent = (props: IProps) => {
+  const {
+    availableVolunteers,
+    socketSend,
+    activeChatIndex,
+    chats,
+    uniqueID,
+  } = useContext(SocketContext);
+  const { isOpen, setIsOpen } = useContext(ModalContext);
+
   const listMessages = () => {
+    /*console.log("Kjører jeg?");
+    console.log(props.messages);
+    console.log(props.messages.map((message, index) => (
+      <ChatMessageComponent key={index} message={message} />
+    )));
+    */
     return props.messages.map((message, index) => (
       <ChatMessageComponent key={index} message={message} />
     ));
   };
 
+  const createFrivilligOptions = () => {
+    return availableVolunteers.map((vol: string) => {
+      console.log(chats[activeChatIndex].roomID);
+      return {
+        inputText: vol,
+        buttonText: 'Legg til',
+        callback: () => socketSend(createJoinChatMessage(
+          chats[activeChatIndex].student,
+          vol,
+          chats[activeChatIndex].messages,
+          chats[activeChatIndex].roomID,
+        )),
+        isDisabled: true,
+      };
+    });
+  };
+
   return (
     <div className="chat-body-container">
+      {isOpen &&
```

Comment:
Bruk lokal state til å tracke om modalen er åpen, ikke state fra modalContext. ModalCOntext skal fjernes helt for å ikke skape problemer. Se master (og tidligere kommentarer 🦀 )